org.opentosca.container.core/src/org/opentosca/container/core/next/repository/SituationsMonitorRepository.java

```diff
@@ -0,0 +1,29 @@
+package org.opentosca.container.core.next.repository;
+
+import java.util.Collection;
+
+import org.opentosca.container.core.next.model.SituationsMonitor;
+
+import com.google.common.collect.Lists;
+
+public class SituationsMonitorRepository extends JpaRepository<SituationsMonitor> {
+
+    public SituationsMonitorRepository() {
+        super(SituationsMonitor.class);
+    }
+    
+    public Collection<SituationsMonitor> findSituationMonitorsBySituationId(Long situationId) {
+        Collection<SituationsMonitor> result = Lists.newArrayList();
+        
+        for(SituationsMonitor moni : this.findAll()) {
+           for(Collection<Long> sits:moni.getNode2Situations().values()) {
+               if(sits.contains(situationId)) {
+                   result.add(moni);
+               }
+           }
+        }
+        
+        return result;
```

Comment:
With streams this turns into:

``` 
return findAll().stream().
  .filter(moni -> moni.getNode2Situations().containsValue(situationId))
  .collect(Collectors.toList());
``` 

Even without streams, using `containsValue` should simplify this a lot.

Reply:
That won't work as situationId is not a collection of longs but could be rewritten anyway